File: PhotoViewer-ObjC/AlbumsPhotosSplitVC.m
//
//  AlbumsPhotosSplitVC.m
//  PhotoViewer
//
//  Created by ASIM27 on 1/20/17.
//  Copyright © 2017 km3h. All rights reserved.
//

#import "AlbumsPhotosSplitVC.h"
#import "AlbumViewController.h"

@interface AlbumsPhotosSplitVC ()

@end

@implementation AlbumsPhotosSplitVC

- (void)viewDidLoad {
    [super viewDidLoad];
    UINavigationController *navigationController = (UINavigationController*)self.viewControllers[0];
    AlbumViewController *albumViewController = (AlbumViewController*)navigationController.topViewController;
    albumViewController.dataSource = self.dataSource;
}

-(void)backButtonTapped
{
    self.preferredDisplayMode = UISplitViewControllerDisplayModePrimaryHidden;
    [self.presentingViewController dismissViewControllerAnimated:YES completion:nil];
}

@end

File: PhotoViewer/AlbumsPhotosSplitVC.swift
//
//  AlbumsPhotosSplitVC.swift
//  PhotoViewer
//
//  Created by ASIM27 on 11/16/16.
//  Copyright © 2016 km3h. All rights reserved.
//

import UIKit

class AlbumsPhotosSplitVC: UISplitViewController
{
    var taskOnline : URLSessionDataTask?
    var taskOffline : URLSessionDataTask?
    public var dataSource : DataSource?
    
    override func viewDidLoad()
    {
        super.viewDidLoad()
        let navigationController = self.viewControllers[0] as! UINavigationController
        let albumViewController = navigationController.topViewController as! AlbumViewController
        albumViewController.dataSource = self.dataSource
    }
    
    public func backButtonTapped()
    {
        self.preferredDisplayMode = .primaryHidden
        self.presentingViewController?.dismiss(animated: true, completion: nil)
    }
}
